In [1]:
import numpy as np
import pandas as pd

In [2]:
df_vendas = pd.read_csv('01.amazon_sales_dataset.csv')
df_vendas.columns

Index(['order_id', 'order_date', 'ship_date', 'delivery_date', 'order_status',
       'customer_id', 'customer_name', 'country', 'state', 'city',
       'product_id', 'product_name', 'category', 'sub_category', 'brand',
       'quantity', 'unit_price', 'discount', 'shipping_cost', 'total_sales',
       'payment_method'],
      dtype='str')

In [3]:
df_vendas.dtypes

order_id              str
order_date            str
ship_date             str
delivery_date         str
order_status          str
customer_id           str
customer_name         str
country               str
state                 str
city                  str
product_id            str
product_name          str
category              str
sub_category          str
brand                 str
quantity            int64
unit_price        float64
discount          float64
shipping_cost     float64
total_sales       float64
payment_method        str
dtype: object

In [4]:
df_vendas.info()

<class 'pandas.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 21 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   order_id        10000 non-null  str    
 1   order_date      10000 non-null  str    
 2   ship_date       10000 non-null  str    
 3   delivery_date   10000 non-null  str    
 4   order_status    10000 non-null  str    
 5   customer_id     10000 non-null  str    
 6   customer_name   10000 non-null  str    
 7   country         10000 non-null  str    
 8   state           10000 non-null  str    
 9   city            10000 non-null  str    
 10  product_id      10000 non-null  str    
 11  product_name    10000 non-null  str    
 12  category        10000 non-null  str    
 13  sub_category    10000 non-null  str    
 14  brand           10000 non-null  str    
 15  quantity        10000 non-null  int64  
 16  unit_price      10000 non-null  float64
 17  discount        10000 non-null  float64
 18

In [24]:
coluna_alvo = 'discount'
# Medidas de Tendencias Centrais 
total_discount = df_vendas[coluna_alvo].sum()
media_valor_discount = df_vendas[coluna_alvo].mean()
discount_minima = df_vendas[coluna_alvo].min()
discount_maxima = df_vendas[coluna_alvo].max()
media_valor_discount

np.float64(0.14996800000000002)

In [27]:
# Medidas de Posição
q1 = np.percentile(df_vendas[coluna_alvo], 25)
#q2 = Mediana
q2 = np.percentile(df_vendas[coluna_alvo], 50)
q3 = np.percentile(df_vendas[coluna_alvo], 75)
q3



np.float64(0.22)

In [28]:
# Avaliar a 'confiabilidade' da média
delta_media_mediana = media_valor_discount - q2
distancia_percentual = media_valor_discount/q2
# Caso 1. Distancia <= 10% (0,9 ~ 1.1)
# R = Média é altamente confiável
#Caso 2. Distancia entre 10% e 25%
#( 0,75 ~ 0.9 / 1.10 ~ 1.25)
# Caso 3. Distancia > 25% (< 0.75 / > 1.25)
# R = NÃO confie na média, ela é distorcida pelos extremos
delta_media_mediana

np.float64(-3.199999999997649e-05)

In [29]:
# Assimetria (Skewness)
assimetria = df_vendas[coluna_alvo]. skew()
assimetria

# Caso 1. Assimetria entre -00,5 e 0,5
# R =  Dados Equilibrados
# Caso 2. Assimetria > 0.5 (Positiva)
# R Valores altos são raros, porém são tão grandes que distorcem o calculo
# Caso 3. Assimetria < - 0.5 (Negativa)
# R  = Menos valores distorcem o calculo para baixo

np.float64(-0.004164539978905712)

In [23]:
print('\n---Resumo Executivo de  discount---')

# Representa o tamanho total da nossa operação no periodo
print(f'2. Gasto Médio:  {media_valor_discount:,.2f}')
# Valor esperado que o cliente gaste em média conosco
print(f'3. Maior Venda:  {discount_maxima:,.2f}')
# Nosso recorde. Investigar o perfil do cliente
print(f'4. Menor Venda:  {discount_minima:,.2f}')
# Menor valor registrado. Investigar se há anomalias ou fraude


---Resumo Executivo de  discount---
2. Gasto Médio:  0.15
3. Maior Venda:  0.30
4. Menor Venda:  0.00


 Comentário
Foi carregada a base 01.amazon_sale_dataset.csv e isolou-se a coluna 'discount'
Segue as medidas centrais:
Média de descontos: 0.1499
Mediana (q2): 0.15
q1: 0.07
q3: 0.22
Assimetria: 0.0041645
Maior Desconto: 0.3
Menor Desconto: 0.0

Avaliação do Delta
Média - Mediana: -3.199* 10(-5)
Média / Mediana: 0.997866

Conclusão: A média é um pouco menor que a mediana. Pode considerar que dados estão perfeitamente distribuídos como um sino. A maioria dos valores está no meio, com poucos extremos iguais para cima e para baixo.
Assimetria Levemente negativa.Distribuição quase simétrica.
A média representa 99% do valor da mediana.
